groups/bsl/bslstl/bslstl_badoptionalaccess.t.cpp

```diff
@@ -0,0 +1,274 @@
+                            "\n======================================\n");
+#ifndef BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
+        const bsl::bad_optional_access b;
+        ASSERT(0 == strcmp("bad_optional_access", b.what()));
+#endif // #ifndef BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
+
+      } break;
+      case 2: {
+        // --------------------------------------------------------------------
+        // 'noexcept' SPECIFICATION
+        //
+        // Concerns:
+        //: 1 The 'noexcept' specification has been applied to
+        //:   'bad_optional_access' constructor as required by the standard.
+        //: 2 The 'noexcept' specification has been applied to
+        //:   'bad_optional_access' 'what' method as required by the standard.
+        // Plan:
+        //: 1 Apply the unary 'noexcept' operator to a 'bad_optional_access'
+        //:   constructor and, for concern 1, confirm that calculated boolean
+        //:   value matches the expected value.
+        //: 2 Apply the unary 'noexcept' operator to 'bad_optional_access'
+        //:   'what' method and, for concern 2, confirm that calculated boolean
+        //:   value matches the expected value.
+        //
+        // Testing:
+        //   CONCERN: Methods qualified 'noexcept' in standard are so.
+        // --------------------------------------------------------------------
+
+        if (verbose) printf("\n'noexcept' SPECIFICATION"
+                            "\n========================\n");
+
+        // N4835: 20.6.5 Class 'bad_optional_access' [optional.bad.access]
+        //..
+        //     namespace std {
+        //       class bad_optional_access: public std::exception {
+        //       public:
+        //         const char* what() const noexcept override;
+        //       };
+        //     } // namespace std
+        //..
+#ifndef BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE
+            == BSLS_KEYWORD_NOEXCEPT_OPERATOR(bsl::bad_optional_access()));
+
+        bsl::bad_optional_access    b;
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE
+            == BSLS_KEYWORD_NOEXCEPT_OPERATOR(b.what()));
+#endif
+      } break;
+      case 1: {
+        // --------------------------------------------------------------------
+        // BREATHING TEST
+        //   This case exercises (but does not fully test) basic functionality.
+        //
+        // Concerns:
+        //: 1 The class is sufficiently functional to enable comprehensive
+        //:   testing in subsequent test cases.
+        //
+        // Plan:
+        //: 1 Perform and ad-hoc test of the primary modifiers and accessors.
+        //
+        // Testing:
+        //   BREATHING TEST
+        // --------------------------------------------------------------------
+
+        if (verbose) printf("\nBREATHING TEST"
+                            "\n==============\n");
+#ifndef BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
```

Comment:
This test is not C++17 specific

Reply:
same as above

Reply:
removed the guards and changed the test to check only that what() returns a non null string.